src/lib.rs

```diff
@@ -1524,6 +1528,59 @@ pub trait Itertools : Iterator {
         combinations::combinations(self, k)
     }
 
+    /// Return an iterator adaptor that iterates over the up to `k`-length combinations of
+    /// the elements from an iterator.
+    ///
+    /// Iterator element type is `Vec<Self::Item>`. The iterator produces a new Vec per iteration,
+    /// and clones the iterator elements.
+    ///
+    /// ```
+    /// use itertools::Itertools;
+    ///
+    /// let it = (1..5).combinations_up_to(3);
+    /// itertools::assert_equal(it, vec![
+    ///     vec![1, 2, 3],
+    ///     vec![1, 2, 4],
+    ///     vec![1, 3, 4],
+    ///     vec![2, 3, 4],
+    ///     vec![1, 2],
+    ///     vec![1, 3],
+    ///     vec![1, 4],
+    ///     vec![2, 3],
+    ///     vec![2, 4],
+    ///     vec![3, 4],
+    ///     vec![1],
+    ///     vec![2],
+    ///     vec![3],
+    ///     vec![4],
+    ///     vec![]
+    /// ]);
```

Comment:
I wonder if the name `combinations_up_to` is perhaps a little misleading, since the combinations are emitted in _descending_ order of size.